staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -545,13 +671,16 @@ func (r *crdHandler) removeDeadStorage() {
 	for uid, crdInfo := range storageMap {
 		if _, ok := storageMap2[uid]; !ok {
 			klog.V(4).Infof("Removing dead CRD storage for %s/%s", crdInfo.spec.Group, crdInfo.spec.Names.Kind)
-			go r.tearDown(crdInfo)
+			go r.tearDown(crdInfo, nil)
 		}
 	}
 }
 
 // Wait up to a minute for requests to drain, then tear down storage
-func (r *crdHandler) tearDown(oldInfo *crdInfo) {
+func (r *crdHandler) tearDown(oldInfo *crdInfo, finishedCh chan<- struct{}) {
```

Comment:
the waitgroup in teardown is based on teh agent making the request and path.  The condition for "let this request through" in this case is based on verb as well.  Verbs doing mutation must be made to fail when the wait starts or we can end up with a late write spoiling the data in storage.